Modify to work with new toolchain, fix x509 decode

earlephilhower opened the pull request:
Several arrays were moved into flash but code was not changed to use
pgm_read_byte everywhere needed.  Fix, avoiding a crash on SSL cert
verification.

Also update makefile for the new libs and toolchain.

earlephilhower committed:
Modify to work with new toolchain, fix x509 decode

Several arrays were moved into flash but code was not changed to use
pgm_read_byte everywhere needed.  Fix, avoiding a crash on SSL cert
verification.

Also update makefile for the new libs and toolchain.

earlephilhower commented:
We already put the update in the Arduino core, so no rush.  Just wanted to keep you in the loop.

igrr commented on Makefile:
```diff
@@ -40,25 +40,14 @@ LDFLAGS  += 	-L$(XTENSA_LIBS)/lib \
-	# Otherwise we need to use a helper function to load 16- and 8-bit constants from program memory.
-    CFLAGS += -DWITH_PGM_READ_HELPER
-endif
-
```
Can this be kept around for the folks who build their toolchain with the -mforce-l32 patch? IIRC this was the case in Sming project.

earlephilhower replied:
Done, with an optional override macro in latest push.

igrr commented on ssl/os_port.h:
```diff
@@ -102,60 +102,15 @@ void ax_wdt_feed();
-    }
-    return 0;
-}
-
```
Likewise, although Arduino is built using a toolchain which includes these functions, it would be great if we can keep the code compatible with the "normal" toolchain as well. Can be done by means of an extra macro set in the makefile.

earlephilhower replied:
Done, by doing a test build in the Makefile to see if the _P functions are already present, and if so set a flag to #ifdef them out.

earlephilhower commented:
I'll see what I can do to clean up the issues you've got later this week.  Thanks for the quick review!

earlephilhower committed:
Use Makefile macros to selectively remove _P functions, -mforce32

Per review request, clean up Makefile to automatically determine if the
stdlibs contain versions of memcpy_P/etc. and bracket them with #ifs in
the code.

Allow users to overrife the mforce32 flag with WITH_PGM_READ_HELPER=1.

earlephilhower committed:
Fix CI failure to run
